pkg/operator/ceph/version/version_test.go

```diff
@@ -0,0 +1,114 @@
+package version
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestToString(t *testing.T) {
+	assert.Equal(t, "14.0.0 nautilus", fmt.Sprintf("%s", &Nautilus))
+	assert.Equal(t, "13.0.0 mimic", fmt.Sprintf("%s", &Mimic))
+	assert.Equal(t, "12.0.0 luminous", fmt.Sprintf("%s", &Luminous))
+
+	expected := fmt.Sprintf("-1.0.0 %s", unknownVersionString)
+	assert.Equal(t, expected, fmt.Sprintf("%s", &CephVersion{-1, 0, 0}))
+}
+
+func TestReleaseName(t *testing.T) {
+	assert.Equal(t, "nautilus", Nautilus.ReleaseName())
+	assert.Equal(t, "mimic", Mimic.ReleaseName())
+	assert.Equal(t, "luminous", Luminous.ReleaseName())
+	ver := CephVersion{-1, 0, 0}
+	assert.Equal(t, unknownVersionString, ver.ReleaseName())
+}
+
+func extractVersionHelper(t *testing.T, text string, major, minor, extra int) {
+	v, err := ExtractCephVersion(text)
+	if assert.NoError(t, err) {
+		assert.Equal(t, v, CephVersion{major, minor, extra})
+	}
+}
+
+func TestExtractVersion(t *testing.T) {
+	// release build
+	v0c := "ceph version 12.2.8 (ae699615bac534ea496ee965ac6192cb7e0e07c0) luminous (stable)"
+	v0d := `
+root@7a97f5a78bc6:/# ceph --version
+ceph version 12.2.8 (ae699615bac534ea496ee965ac6192cb7e0e07c0) luminous (stable)
+`
+	extractVersionHelper(t, v0c, 12, 2, 8)
+	extractVersionHelper(t, v0d, 12, 2, 8)
+
+	// development build
+	v1c := "ceph version 14.1.33-403-g7ba6bece41"
+	v1d := `
+[nwatkins@smash build]$ bin/ceph --version
```

Comment:
you might remove your alias from the output

Reply:
alias?

Reply:
just referring to `nwatkins`

Reply:
Oh, sure. :)